### PR TITLE
feat: allow configuring public/private tenant LB

### PR DESCRIPTION
**Pull request by mjnagel**
Changes:
- Adds a `private_tenant_lb` zarf var to configure whether the provisioned tenant elastic loadbalancer is public or private
- Adds actions before/after to cycle the tenant loadbalancer service if this value changes on an upgrade (ensures a new ELB is provisioned)

Notes:
- all logic was added to the AWS package as it is currently AWS specific but we could move to "core" DUBBD in the future if other flavors require the same variable/functionality
- bash scripting is not intended to be the final solution for "upgrade logic" - there are other proposed routes of doing things in a more programmatic approach w/testing, etc but as there is not a clear decision on that approach yet this was done with basic scripting
- this basic pattern would likely apply to all future upgrade logic as well - `onDeploy.before` actions to check before state, then a final component to take action based on said state

Closes https://github.com/defenseunicorns/uds-package-dubbd/issues/297

### CODE REVIEW

File: aws/values/aws-istio.yaml [hunk 1]
```diff
@@ -3,10 +3,10 @@ istio:
     admin-ingressgateway:
       kubernetesResourceSpec:
         serviceAnnotations:
-          service.beta.kubernetes.io/aws-load-balancer-internal: "true"
+          service.beta.kubernetes.io/aws-load-balancer-internal: "###ZARF_VAR_PRIVATE_TENANT_LB###"
```
Comment (runyontr) [inline aws/values/aws-istio.yaml:6]: shouldn't this be a different variable?  i.e. `###ZARF_VAR_PRIVATE_ADMIN_LB###`?

Comment (mjnagel) [reply aws/values/aws-istio.yaml:6]: Oof - definitely messed this up. I intended to leave this one as hardcoded `"true"`, and not allow configuration on it but open to other's opinions. Thanks for catching - if we want it configurable I'll make it a separate var and ensure the upgrade "job" will handle config changes as well.

**Comment by mjnagel** [comment]
Noting that this does not provide docs/configuration of external access - that will be done as a part of https://github.com/defenseunicorns/uds-package-dubbd/issues/71

**Review by MxNxPx — APPROVED** [review]
lgtm :rocket:

**Pull request merged**